index.js

```diff
@@ -62,17 +74,68 @@ module.exports = {
         */
         'no-loop-func': 'error',
 
+        /*
+            Enforce parenthesis in cases of mixed precedence.
+        */
+        'no-mixed-operators': ['error', {
+            groups: [
+                ['%', '**'],
+                ['%', '+'],
+                ['%', '-'],
+                ['%', '*'],
+                ['%', '/'],
+                ['/', '*'],
+                ['&', '|', '<<', '>>', '>>>'],
+                ['==', '!=', '===', '!=='],
+                ['&&', '||'],
+            ],
+            allowSamePrecedence: true
+        }],
+
         /*
             This rule defends against prototype pollution vulnerabilities that exploitable with
             XSS.
         */
         'no-prototype-builtins': 'error',
 
+        /*
+            Higher-order function composition is preferred for iterative loops. For...in statements
+            can be used when performance is a consideration as long as guard-for-in is enabled.
+            For..of statements can be used when break is required for control flow or the iteration
+            produces side-effects.
+        */
+        'no-restricted-syntax': [
+            'error', {
+                selector: 'LabeledStatement',
+                message: 'Labels are a form of GOTO; using them makes code confusing and hard to maintain and understand.',
+            }, {
+                selector: 'WithStatement',
+                message: '`with` is disallowed in strict mode because it makes code impossible to predict and optimize.',
+            }],
+
         /*
             Migrating from var is not trivial, so there may be exceptions for older projects.
         */
         'no-var': 'error',
 
+        /*
+            Defining classes and variables before their use can cause errors. However, defining
+            functions and methods before their use is a common programming practice for
+            readability.
```

Comment:
"defining functions" is a little vague, ie a variable binding to a function expression is defining a function and would not be allowed. Also the position as written seems backwards from the decision for functions.

Do you think the following may be clearer?: "However, placing function declarations at the end of a file is common programming practice for readability"

Reply:
Whoops! Yes, that is fine.